apps/site/lib/site/shuttle_diversion.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Site.ShuttleDiversion do
+  @moduledoc "Represents and retrieves data about rail-replacement shuttle bus services."
+
+  alias Alerts.Match, as: AlertsMatch
+  alias Alerts.Repo, as: Alerts
+  alias Routes.Route
+  alias Stops.Repo, as: Stops
+  alias V3Api.Trips
+
+  defmodule Shape do
+    @moduledoc "Derivative of an API `shape` tailored for drawing a shuttle diversion map."
+
+    @enforce_keys [:id, :direction_id, :polyline, :is_shuttle_route]
+    defstruct @enforce_keys
+
+    @type t :: %__MODULE__{
+            id: String.t(),
+            direction_id: 0 | 1,
+            polyline: [[float]],
```

Comment:
Question - should this indicate a list of lat/lng positions? Or is this the encoded version?

Reply:
This is the decoded polyline, which is a list of lists of floats. Technically the lists of floats always contain exactly 2 elements (lat and long), but there is no way to encode that in the typespec. The idea is that if you have a fixed-length list, what you actually want is a tuple, and that _can_ be encoded as `{float, float}`. However, tuples can't be JSON-encoded.